Plugins/MatrixPlugin/MatrixLogic.cs

```diff
@@ -0,0 +1,434 @@
+            _character.Gear.CollectionChanged += Gear_CollectionChanged;
+            Actions = matrixActions;
+            ProgramsList = programs;
+
+            //Load all CyberDecks,Commlinks and Programs to the Lists
+            AddEquipment(character.Gear);
+        }
+
+        #region HelperFunctions
+
+        private void AddEquipment(IList equipment)
+        {
+            foreach (Gear gear in equipment)
+            {
+                if (gear.Category == "Cyberdecks" || gear.Category == "Commlinks")
+                    Persons.Add(gear);
+                else if (gear.Category.Contains("Program"))
+                {
+                    Program newItem = ProgramsList.Find(x => x.Name.Equals(gear.Name));
+                    newItem.logic = this;
+                    Programs.Add(newItem);
+                }
+                if (gear.Children.Count > 0)
+                    AddEquipment(gear.Children);
+            }
+        }
+
+        private void RemoveEquipment(IList equipment)
+        {
+            foreach (Gear gear in equipment)
+            {
+                if (gear.Category == "Cyberdecks" || gear.Category == "Commlinks")
+                    Persons.Remove(gear);
+                else if (gear.Category.Contains("Program"))
+                    Programs.Remove(ProgramsList.Find(x => x.Name.Equals(gear.Name)));
+            }
+        }
+
+        private void Gear_CollectionChanged(object sender, System.Collections.Specialized.NotifyCollectionChangedEventArgs e)
+        {
+            if (e.NewItems != null)
+                AddEquipment(e.NewItems);
+            if (e.OldItems != null)
+                RemoveEquipment(e.OldItems);
+        }
+
+        private void character_PropertyChanged(object sender, PropertyChangedEventArgs e)
+        {
+            if (e.PropertyName == nameof(Character.WoundModifier))
+            {
+                OnPropertyChanged(nameof(WoundModifier));
+            }
+        }
+
+        public int GetTotalMatrixAttribute(string attribute)
+        {
+            switch (attribute)
+            {
+                case "Attack":
+                    return TotalAttack;
+                case "Sleaze":
+                    return TotalSleaze;
+                case "Data Processing":
+                    return TotalDataProcessing;
+                case "Firewall":
+                    return TotalFirewall;
+            }
+            return 0;
+        }
+
+        public int GetTotalSkill(string skillName)
+        {
+            Skill skill = _character.SkillsSection.GetActiveSkill(skillName);
+            if (skill != null)
+                return skill.TotalBaseRating;
+            return 0;
+        }
+
+        public int GetTotalAttribute(string attributeName)
+        {
+            CharacterAttrib attribute = _character.GetAttribute(attributeName);
+            if (attribute != null)
+                return attribute.TotalValue;
+            return 0;
+        }
```

Comment:
Note: this *will* misbehave if the character also has an improvement that is supposed to swap the attribute being used for a particular skill that happens to be a Matrix skill. You need to be smarter about this and think more in terms of what could possibly be done within the pool of possible adjustments and bonuses and house rules, because people can and do and will do all sorts of weird things. This applies doubly so with Matrix-related stuff because 5e's Matrix rules are one of the most frequent targets of house rules.